docs/external-node.md

```diff
@@ -0,0 +1,305 @@
+
+```yaml
+kind: NetworkPolicy
+metadata:
+  name: anp1
+  namespace: vm-ns
+spec:
+  priority: 9000.0
+  appliedTo:
+    - externalEntitySelector:
+        matchLabels:
+          role: db
+  egress:
+    - action: Reject
+      ports:
+        - protocol: TCP
+          port: 22
+      to:
+        - externalEntitySelector:
+            matchLabels:
+              role: front
+        - ipBlock:
+            cidr: 172.16.100.5/32
+```
+
+`antrea-agent` running on the target external Node sets up a connection with
+`antrea-controller` using the `ExternalNode` name as an option to receive the internal Antrea NetworkPolicy objects.
+More details please refer to the
+[installation guide](vm-installation.md) for `antrea-agent` on an external Node.
+
+`antrea-agent` realizes the Antrea NetworkPolicy on OVS using OpenFlow entries.
+
+## Security
+
+An external Node is thought as an untrusted point on the network. To follow the security principle that least
+permissions are given, the RBAC for `antrea-agent`
+running on `ExternalNode` feature is modified:
+
+- Only `get`, `list` and `watch` permissions are given on resource `ExternalNode`
+- Only `update` permission is given on resource `antreaagentinfos`, and `create`
+  permission is moved to `antrea-controller`
+- for more details please refer to [vm-agent.yaml](../build/yamls/externalnode/vm-agent.yaml)
+
+`antrea-agent` reports its status by updating the `antreaagentinfo` resource which is configured with the same name as
+the `ExternalNode`. `antrea-controller`
+creates an `antreaagentinfo` object for each new `ExternalNode`, and then `antrea-agent`
+updates it every minute with its latest status. `antreaagentinfo` is deleted by
+`antrea-controller` when the `ExternalNode` is removed.
+
+## Traffic model
+
+### OVS pair ports
+
+`antrea-agent` uses the interface IPs or name to find the network interface on the external Node, and then attaches it
+on the OVS bridge. The network interface is attached on OVS as uplink, and a new OVS internal Port is added to take over
+the uplink interface's IP/MAC and routing configurations. On Windows, the DNS configurations are also moved to the OVS
+internal port from uplink. Before attaching the uplink to OVS, the network interface is renamed with a suffix
+"~", and OVS internal port is configured with the original name of the uplink. As a result, IP/MAC/routing entries are
+seen on a network interface configuring with the same name on the external Node.
+
+The outbound traffic sent from the external Node enters OVS from the internal port, and finally output from the uplink,
+and the inbound traffic enters OVS from the uplink and output to the internal port. The IP packet is processed by the
+OpenFlow pipeline, and the non-IP packet is forwarded directly.
+
+Following is a diagram for the traffic model on an external Node.
+![Traffic On ExternalNode](assets/traffic_external_node.svg)
+
+### OpenFlow pipeline
```

Comment:
This section is not interested by most users. I think we should move it to a separate section.

Reply:
updated.